### PR TITLE
Bloody Bastard Sword Removal

### PR DESCRIPTION
**Pull request by Nilons**
Partial revert of #30887 

:cl:
del: Bloody bastard cult sword 
/:cl:

[why]: People have been complaining about it since it was put in, its broken, robustin knew it was broken when he pr'd it considering the spin that kills everything in a 20km radius is named spin2win in the code

### CODE REVIEW

**Comment by Nilons** [comment]
I think the stupid cqc thing I did a long time ago also got thrown in because Im retarded and dont know to code or pr so if it did can someone delete that

File: code/modules/uplink/uplink_item.dm
Comment (vuonojenmustaturska) [inline code/modules/uplink/uplink_item.dm]: what's this

**Comment by BeeSting12** [comment]
you probably based if off an older branch

**Comment by Nilons** [comment]
I tried really hard to not do that god DAMMIT

**Comment by vuonojenmustaturska** [comment]
it's fine now when it gets squashed anyways

**Comment by vuonojenmustaturska** [comment]
apart from the part where this doesn't compile

**Comment by Robustin** [comment]
The whirlwind is a melee range only and it only goes off once every ~1.5s (slower than a standard melee).

It's been around for a week and probably shown up in less than 5 rounds. I've only heard one piece of feedback and it was 2nd-hand from Kor that someone had built 7 forges, made 7 swords, and steamrolled the station - which is pretty impressive, I've never seen that much coordination from a cult before.

Obviously I'm keeping an eye on the balance here but cult's win rate  last month was in the low 30's and I'm not convinced there's a problem until these weapons start appearing regularly or cult's winrate skyrockets. 

This is the definition of a kneejerk PR and you can't even offer a coherent explanation of why it's appropriate.

**Comment by Nilons** [comment]
It's so broken an admin actually approached me and jokingly asked me to pr a removal despite me never having coded or done a real pr before.  Delimbing everyone around every 1.5s while deflecting all of their attacks is not balanced.  Being able to get a colossus down to half health standing next to it naked because of the parry and deflection is not balanced.

**Comment by Qbopper** [comment]
>coders
>playing

nilons cmon

**Comment by Anonmare** [comment]
I think I was in on that round and it was only me and another guy who outfitted the cult with empowered robes and bastard swords. We were both engies and made a spacebase and ~15 forges using plasteel from engineering and EVA to help make more runed metal. 

Whilst they were making forges and the swords, I went ahead and backdoored into security with a teleport rune and when we were all ready, summoned everyone to the base, got them armour and a sword and teleported them into perma whereupon we smashed our into security proper, jaunted into the warden's office and killed him in seconds, then forced our way into the armoury and from that point on nobody could mount a defense against us.

My biggest pointer would be to make the jaunt blocked by structures because if we had LoS, we could literally teleport behind people and nothing personnel them without a sweat. The sword also ought to have some other downside (like slower run speed or anything that lets victims have a chance to get away) because a single cultist with one in empowered armour is a threat to an entire security team by their lonesome, and in a group they're overpowering

**Review by Cyberboss — APPROVED** [review]

**Comment by Robustin** [comment]
This all sounds like it was inspired by a round where they were mass produced, like I said I'm keeping an eye out because I genuinely didn't expect them to get made en-masse. Most cults don't even make structures, and I have trouble getting the cult to even stand around a sac rune with our target, let alone striking the right balance of conversion + base-making to have several swords ready, the converts to use them, and the organization to have those cultists show up, equip them, then stick together. 

I've got a dozen levers I can pull to nerf or buff this thing but we're not even close to seeing where the meta shakes out on this.

**Comment by ChangelingRain** [comment]
You should probably stop balancing around the structures being a sometimes thing, since they massively affect winrate.

**Comment by Nilons** [comment]
I didn't know about that round when I made the PR, I only saw admins spawning one at a time in the thunderdome and doing ridiculous stuff with it.  The colossus example was not an exaggeration, I watched DGL do that.  Then him and beesting killed an entire crowd of wanderers, so called toughest enemy that does the most damage without taking a hit using them.

**Comment by Jordie0608** [comment]
@Robustin The whirlwind ability procs every 0.8 seconds (ticklag notwithstanding)
https://github.com/tgstation/tgstation/blob/9b7f1ada6be169665f7e4397582d09cdc8c9f735/code/datums/status_effects/buffs.dm#L387-L390
https://github.com/tgstation/tgstation/blob/32050bed011e588739a344da3c0663b6c1b65ca3/code/datums/status_effects/status_effect.dm#L8

**Comment by Robustin** [comment]
@ChangelingRain 

Structures influence on winrate hasn't been my experience, only the veil shifter (and im literally the only person I've seen use the shifter) and shuttle curse have had enough of an impact to tip the outcome of rounds for me. 

If that changes I'd consider moving some of the utility back to talismans but structures remain a choice that involves a nuanced cost/benefit decision. 

My biggest issue with the structures right now is the fact that they are almost impossible to use effectively on station and are probably one of the biggest drivers in space-basing. The lighting (that you added) means it's suicidal to put them anywhere in maint because literally the first person to walk by is going to notice that some dark corner of maint is now glowing red.

I'm tempted to have runes and structures automatically disable nearby cameras because with the AI being able to see runes again it's downright oppressive trying to establish a cult base.

**Comment by Nilons** [comment]
Jordie literally caught you lying right there, why are you defending this so hard.

**Comment by Jordie0608** [comment]
Misinformation =/= lying m8, plz no aggravate discussion.

**Comment by Robustin** [comment]
Because @Jordie0608 hasn't tested this. My experience in testing was that the standard tick rate was the floor for updating buffs and the 8 value I used didn't do anything. Considering you have no first-hand experience with this weapon it seems absurd for you to play the "you dont know what you're talking about" card.

**Comment by ChangelingRain** [comment]
The sword actually does hit every 0.8 seconds, because that's the point of status effects being on fastprocess; they're fast and more accurate than any other ticker.

**Comment by Nilons** [comment]
It's not really me pulling the you don't know what you're talking about card considering two more experienced coders have agreed you are wrong, I chalked the misinformation up to deliberate lying because I have not had an interaction with robustin where he didn't lie constantly.  Which may have been jumping the gun a bit you're right jordie

**Comment by Robustin** [comment]
I've just tested it.

The whirlwind status is 50 deciseconds, if it was really proc'ing every 8 deciseconds then you'd get 6 hits.

I always got 5 hits with the whirlwind, always and keep in mind there isn't an immediate first hit... there's a full second of spinning before the first attack goes out. 

I regret even engaging in this discussion when you're opening a kneejerk revert based on admins killing NPC's on Centcom. For simple mobs the sword is basically a worse double esword (dbl has a 75% blockrate to this sword's 50%, you will take twice as many hits when using this sword). 

One hit per second is faster than I thought but I guess one second just feels slow to me - my original point stands that the whirlwind is still slower than just rapidly clicking on someone. **I also verified during testing that you cannot attack during the spin so 35DPS is the absolute maximum damage you will do to someone while spinning.**

**Comment by sawrge** [comment]
Forever lol at Robustin balance

**Comment by sawrge** [comment]
Boy defo has some kinda autism it's all just numbers numbers numbers like reading Dota patch notes or smth

**Comment by sawrge** [comment]
[
30406412-a4048428-98bf-11e7-876d-053b3f24d23f
](url)

**Comment by DoctoreMad** [comment]
>its only 35DPS

Haha shit

**Comment by Nilons** [comment]
35 DPS not taking into account bloodloss, delimbing, and the fact that wielding it makes you basically unkillable

**Comment by Iamgoofball** [comment]
Just limit to 1 in a round.

**Comment by SyruseEln** [comment]
This is the second time I have ever agreed with one of goofballs opinions.

**Comment by Darkgenerallord** [comment]
Just make it a cult master exclusive they can summon or something tbh

**Comment by Robustin** [comment]
Again, id consider all these things but jesus I haven't even seen one in a real round yet. Slow your roll. 

Also delimbing is technically a damage reducer, not increaser. I've actually been healed by losing limbs before. Yes 35 isn't something to sneeze at but I was just illustrating its the same as any other heavy-duty melee weapon, even with berserker robes on you are slowing than a healthy human while spinning... if the victim doesn't use the whirlwind's initial delay to run away, they kinda deserve what comes next.

**Comment by Jordie0608** [comment]
In my test:
>start spinning at time 1181 with interval 8 and duration 50 until 1231
sword spin for tick at 1189 next at 1197, compensated interval since last tick is 8
sword spin for tick at 1197 next at 1205, compensated interval since last tick is 8
sword spin for tick at 1206 next at 1214, compensated interval since last tick is 9
sword spin for tick at 1214.5 next at 1222.5, compensated interval since last tick is 8.5
sword spin for tick at 1223.5 next at 1231.5, compensated interval since last tick is 9

Which confirms @Robustin's findings: the whirlwind procs every .8 seconds but experiences enough ticklag to stop a 6th hit from going off.

**Comment by Qbopper** [comment]
why do the numbers matter if fighting it is hilariously retarded anyways, though

**Comment by Bawhoppen** [comment]
why so many walls of text

**Comment by Dorsisdwarf** [comment]
The combination of "special spinning delimbing whirlwind of immunity" and 40 brute and insta-shard on down and teleporting right behind people through all barriers in _one_ mass-produceable item is farcial and I dont know what you were thinking. This should have been a minimum of two different items

**Comment by Incomptinence** [comment]
Probably the jaunt if the problem more than anything there is a reason we keep teleport spells to wizards and cults take the subspaceway.

Doesn't matter if the damage of the whirlwind is worse than just clicking on someone if you literally teleported in and cornered them while their limbs fly off.

**Comment by DoctoreMad** [comment]
>Robustin's hardware is too slow so fastprocess gets MC throttled resulting in a slower FP than live server.

haha shit

**Comment by ExcessiveUseOfCobblestone** [comment]
> why do the numbers matter if fighting it is hilariously retarded anyways, though

who made the feature again?

**Comment by optimumtact** [comment]
Don't assume you won't get six hits in just because you couldn't do it on your machine

Machines are different, and you should always "assume" that the code will run at the stated tick rate, even though in reality it's not likely to be accurate

**Comment by KorPhaeron** [comment]
Also this arguing about DPS and whatever is silly since it only takes a single hit to remove the guys leg, at which point he can't run and you are invincible so whether it kills him in 2 seconds or 4 seconds doesn't really matter

**Comment by KorPhaeron** [comment]
All the theorycrafting aside everyone who has actually used the sword in game seems to agree it needs to be limited or weaker or something

**Comment by KorPhaeron** [comment]
Letting him try and balance it seems more fair than outright removal within the first couple days though

**Comment by Nilons** [comment]
the removal was because he wasn't trying to balance it as far as I knew and I wasn't gonna do it for him.  Leaving something broken that ruins the round when people play cult in the game until robustin gets around to fixing it is gonna be a while

**Comment by Mercenaryblue** [comment]
Here's an idea. If an item turns you into a spinning, unstoppable AoE murder machine, maybe, just maybe, limit its proc to once every 5 seconds.

**Comment by Robustin** [comment]
@Mercenaryblue 

Do you even understand what the words you're using mean?

**Comment by DoctoreMad** [comment]
@Robustin He does.

**Comment by Cyberboss** [comment]
Stale, conflicts, @Robustin nerf this

**Pull request closed without merging**
